### PR TITLE
updated dashboard and referal module- 17feb

### PR DESCRIPTION
**Pull request by gem-divyamadan**
Hi Neeraj,
Please get the code reviewed for two modules.
1.DashboardProfile
2.Referral
Thanks

### CODE REVIEW

**Comment by gem-divyamadan** [comment]
Hi Neeraj,
Please review the code for Dashboard profile and referral.
Thanks

File: src/main/java/com/qa/mis/stepdefinition/ReferalStep.java [hunk 1]
```diff
@@ -31,83 +32,108 @@ public void enterPassword() {
...
+
+            try {
+                //  DriverAction.waitUntilElementAppear(popUpCloseButton, 7);
+                DriverAction.waitSec(5);
+                DriverAction.click(popUpCloseButton, "close button");
+                // DriverAction.click(clickOnChangeDetails,"change details button");
+            } catch (Exception e) {
+                // e.printStackTrace();
+                GemTestReporter.addTestStep("ERROR", "SOME ERROR OCCURRED" + e, STATUS.FAIL, DriverAction.takeSnapShot());
+            }
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/ReferalStep.java:56]: GemTestReporter.addTestStep("ERROR", "SOME ERROR OCCURRED" + e, STATUS.FAIL, DriverAction.takeSnapShot());

Please use proper error definition

File: src/main/java/com/qa/mis/stepdefinition/ReferalStep.java [hunk 2]
```diff
@@ -116,51 +142,143 @@ public void referralWindowIsVisible() {
...
         }
     }
 
     @And("^click on email under referral$")
     public void clickOnEmailUnderReferral() {
-        DriverAction.waitUntilElementAppear(refEmail,3);
-        DriverAction.click(refEmail,"referral email text");
+        //DriverAction.waitUntilElementAppear(refEmail, 3);
+        DriverAction.waitSec(5);
+        DriverAction.click(refEmail, "referral email text");
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/ReferalStep.java:162]: either verify of surround selenium click function in try catch statement for every instance

**Review by Gem-NeerajSingh — CHANGES_REQUESTED** [review]
Please update code as per given comments

File: src/main/java/com/qa/mis/stepdefinition/ReferalStep.java (hunk 2 above)
Comment (gem-divyamadan) [reply src/main/java/com/qa/mis/stepdefinition/ReferalStep.java:162]: Done pls have look.!

**Comment by gem-divyamadan** [comment]
Changes made

**Comment by gem-divyamadan** [comment]
Done changes 23feb23

**Review by Gem-NeerajSingh — APPROVED** [review]

File: src/main/resources/features/Referal.feature [hunk 3]
```diff
@@ -14,17 +14,33 @@ Feature: MIS
...
-        Scenario: Validation of wrong email pop up
+  Scenario: Validation of wrong email pop up
           When user checks referral section
           And user click on referral action
           And click on email under referral
           And enter the email address
           And click on save
           Then verify the warning message in ref section
 
+          Scenario: Validation of wrong format upload pop
```
Comment (Gem-NeerajSingh) [inline src/main/resources/features/Referal.feature:30]: pop up

Comment (gem-divyamadan) [reply src/main/resources/features/Referal.feature]: Hi Neeraj,
Pop up done & one locator added in dashboard.
Thanks

**Review by Gem-NeerajSingh — APPROVED** [review]

File: src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java [hunk 4]
```diff
@@ -29,7 +34,8 @@ public void enterPassword() {
 
     @And("^click on SignIn button$")
     public void signIn() {
-        DriverAction.waitUntilElementAppear(signIn, 2);
+        // DriverAction.waitUntilElementAppear(signIn, 2);
+        DriverAction.waitSec(5);
         DriverAction.click(signIn);
         //DriverAction.waitSec(10);
```
Comment (Gem-NeerajSingh) [inline src/main/java/com/qa/mis/stepdefinition/DashboardProfileSteps.java:44]: Please Remove Commented Code

**Pull request closed without merging**